### PR TITLE
[OPS] Implement Various improvements in mainstream deployment templates

### PR DESCRIPTION
**Pull request by i5okie**
- Implement template changes from previous various improvements pr
- Update GitHub pipelines
- Do not cache frontend dependencies past Develop
- Tune Document Manager resources and health checks

### CODE REVIEW

File: infrastructure/prime-app-ephemeral-template.yml [hunk 1]
```diff
@@ -1268,8 +1258,8 @@ objects:
                   name: document-manager
             resources:
               limits:
-                cpu: 25m
-                memory: 198Mi
+                cpu: 80m
+                memory: 200Mi
```
Comment (itchison) [inline infrastructure/prime-app-ephemeral-template.yml:1262]: Good catch. 198Mi wasn't enough.

File: infrastructure/prime-app-template.yml [hunk 2]
```diff
@@ -999,9 +977,24 @@ objects:
           - name: ${SVC_NAME}-redis-data
             emptyDir: {}
         containers:
-          - resources:
+          - name: redis
+            image: >-
+              image-registry.openshift-image-registry.svc:5000/${OC_LICENSE_PLATE}-tools/redis-6:latest
```
Comment (itchison) [inline infrastructure/prime-app-template.yml:982]: Latest?

Comment (itchison) [inline infrastructure/prime-app-template.yml:982]: Maybe consistency-  but I also see a lot of ${{secrets.OPENSHIFT_IMAGE_REPOSITORY}} in these PRs. Should we have them all the same?

File: .github/workflows/build-main.yml [hunk 3]
```diff
@@ -168,14 +197,10 @@ jobs:
...
-        env:
-          COMPOSE_DOCKER_CLI_BUILD: 1
-          DOCKER_BUILDKIT: 1
-
         run: |
           oc process -f infrastructure/prime-app-template.yml \
-            -p ASP_ENV=Development \
-            -p VANITY_URL=pharmanetenrolment.gov.bc.ca \
+            -p ASP_ENV=Production \
+            -p VANITY_URL=prod.pharmanetenrolment.gov.bc.ca \
```
Comment (itchison) [inline .github/workflows/build-main.yml:203]: Is this url correct? "prod" ?

File: infrastructure/prime-app-template.yml (hunk 2 above)
Comment (i5okie) [reply infrastructure/prime-app-template.yml:982]: yes, it's an image in our tools namespace.

Comment (i5okie) [reply infrastructure/prime-app-template.yml:982]: `secrets.OPENSHIFT_IMAGE_REPOSITORY` is a GitHub actions-only thing. Having it in the OpenShift will make them anywhere else.

Comment (itchison) [reply infrastructure/prime-app-template.yml:982]: Ah I see, we control it what version is in there.

**Review by itchison — APPROVED** [review]

**Review by hogmann — APPROVED** [review]
Nothing looks, though the pr doesn't seem to load but that might be because of memory issues.

**Pull request merged**